addons/auth_totp_mail_enforce/data/mail_template_data.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
+        <record id="mail_template_totp_mail_code" model="mail.template">
+            <field name="name">TOTP for users: Authentication by email</field>
+            <field name="model_id" ref="base.model_res_users" />
+            <field name="subject">Your two-factor authentication code</field>
+            <field name="partner_to">{{ object.partner_id.id }}</field>
+            <field name="lang">{{ object.partner_id.lang }}</field>
+            <field name="auto_delete" eval="True"/>
+            <field name="body_html" type="html">
+<div style="margin: 0px; padding: 0px; font-size: 13px;">
+    <p style="margin: 0px; padding: 0px; font-size: 13px;">
+        Dear <t t-out="object.partner_id.name or ''"></t><br/><br/>
+        <p>Someone is trying to log in into your account with a new device.</p>
+        <ul>
+            <li>Location: <t t-out="ctx.get('location') or 'N/A'"/></li>
+            <li>Device: <t t-out="ctx.get('device') or 'N/A'"/></li>
+            <li>Browser: <t t-out="ctx.get('browser') or 'N/A'"/></li>
+            <li>IP address: <t t-out="ctx.get('ip') or 'N/A'"/></li>
+        </ul>
+        <p>Please enter the following code to complete the login:</p>
+        <t t-set="code_expiration" t-value="object._get_totp_mail_code()"/>
+        <t t-set="code" t-value="code_expiration[0]"/>
+        <t t-set="expiration" t-value="code_expiration[1]"/>
+        <div style="margin: 16px 0px 16px 0px; text-align: center;">
+            <span t-out="code" style="background-color:#faf9fa; border: 1px solid #dad8de; padding: 8px 16px 8px 16px; font-size: 24px; color: #875A7B; border-radius: 5px;"/>
+        </div>
+        <small>Please note that this code will expire on <t t-out="format_datetime(expiration, dt_format='long', tz=object.partner_id.tz, lang_code=object.partner_id.lang)"/>.</small>
```

Comment:
This may be a little bit misleading, due to the timezone conversion and the rounding rules for the TOTP check.

Example: if current time is `03:01:00 UTC`, `expiration` is going to be `04:01:00 UTC`. But in reality the code will be valid almost 1 hour more (until 04:59:59 UTC), due to the rounding logic.

Also we may rely on a wrong TZ setting, which could become very confusing for the end-user.

How about we don't mention an exact expiration date in the email, and simply tell the user that the code will be valid for "an hour". Even if there is a delay of 5 minutes for the email reception, the expectations of the user will be correct: use the code asap, it won't be valid for very long.

Reply:
Done : ) Using babel in case we do change to make the expiration delta customizable, and to have the automatic translation.